File: Numeros Pseudo-Aleatorios/Linear Congruential/LinearCongruential.cpp
/* 
 PRNG .......: Linear Congruential
 Ano ........: 1965
 Autor ......: W. E. Thomson
*/

#include <iostream>
#include <iomanip>
#include <ctime>

double LinearCongruential(long long& seed) {
    const long long M = 2147483647;
    seed = (1664525 * seed + 1013904223) % M;
    return static_cast<double>(seed) / M;
}

int main(void) {
  
    long long Xn1 = 1234;
    long long Xn2 = static_cast<long long>(std::time(nullptr));

    std::cout << "GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)\n";
    std::cout << "PSEUDORANDOM NUMBER GENERATOR (PRNG)\n";
    std::cout << "PADRAO: Linear Congruential\n";
    std::cout << "==========================================\n\n";
    std::cout << "    Semente: 1234        Semente: TIME\n";
    std::cout << "    -------------        -------------\n";

    for (int i = 1; i <= 10; i++) {
        std::cout.precision(11);
        std::cout << std::fixed << std::setw(17);
        std::cout << LinearCongruential(Xn1) << "        ";
        std::cout << std::setw(13);
        std::cout << LinearCongruential(Xn2) << std::endl;
    }

    return 0;
}

File: Numeros Pseudo-Aleatorios/Linear Congruential/LinearCongruential.py
"""
   PRNG .......: Linear Congruential
   Ano ........: 1965
   Autor ......: W. E. Thomson
"""

import math
import time

def linearCongruential(seed):
    M = 2147483647
    seed[0] = (1664525 * seed[0] + 1013904223) % M
    return seed[0] / M

def main():
    Xn1 = [1234]
    Xn2 = [int(time.time() * 1000)]

    print("GERADOR DE NUMEROS PSEUDO ALEATORIOS (GNA)")
    print("PSEUDORANDOM NUMBER GENERATOR (PRNG)")
    print("PADRAO: Linear Congruential")
    print("==========================================")
    print()
    print("    Semente: 1234        Semente: TIME")
    print("    -------------        -------------")

    for i in range(1, 11):
        print("{:17.11f}        {:13.11f}".format(linearCongruential(Xn1), linearCongruential(Xn2)))

if __name__ == "__main__":
    main()
